### PR TITLE
Replace "project archived" icon

### PR DESCRIPTION
**Pull request by vitali-l2beat**
Resolves L2B-937
https://linear.app/l2beat/issue/L2B-937/replace-project-archived-icon-on-details-page

### CODE REVIEW

File: packages/frontend/src/components/project/Callout.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export function Callout({ color, icon, body, className }: CalloutProps) {
         className,
       )}
     >
-      <span>{icon}</span>
+      <span className={cx('flex items-center')}>{icon}</span>
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/Callout.tsx:24]: Changes in this component caused many undesired layout shifts (check out the storybook-visual-regression). IMO, we could use a new simple component with a background and icon, that will be only responsible for the "project archived" warning. No need to complicate the warning bar then.

**Review by michalsidzej — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/project/WarningBar.tsx [hunk 2]
```diff
@@ -26,7 +29,13 @@ export function WarningBar({
     <Callout
       className={cx('p-4', className)}
       color={color}
-      icon={<ShieldIcon className={cx(iconFill)} />}
+      icon={
+        isArchived ? (
+          <ArchivedIcon className={cx()} />
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/WarningBar.tsx:34]:
```suggestion
          <ArchivedIcon />
```

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import React from 'react'
+
+import { ArchivedIcon } from '../icons/symbols/ArchivedIcon'
+
+export interface ArchiveBarProps {
+  text: string
+  className?: string
+}
+
+export function ArchivedBar({ text, className }: ArchiveBarProps) {
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:11]:
```suggestion
export function ArchivedBar({ text }: ArchiveBarProps) {

```

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import cx from 'classnames'
+import React from 'react'
+
+import { ArchivedIcon } from '../icons/symbols/ArchivedIcon'
+
+export interface ArchiveBarProps {
+  text: string
+  className?: string
+}
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:9]:
```suggestion
export interface ArchiveBarProps {
  text: string
}

```

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+export interface ArchiveBarProps {
+  text: string
+  className?: string
+}
+
+export function ArchivedBar({ text, className }: ArchiveBarProps) {
+  return (
+    <div
+      className={cx('bg-gray-750 flex justify-center rounded-lg', className)}
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:14]:
```suggestion
      className='bg-gray-750 flex justify-center rounded-lg'

```

File: packages/frontend/tailwind.config.js [hunk 6]
```diff
@@ -31,6 +31,7 @@ module.exports = {
         600: '#848484',
         650: '#5C5C5C',
         700: '#565656',
+        750: '#424045',
         800: '#424242',
```
Comment (michalsidzej) [inline packages/frontend/tailwind.config.js:35]: Are you sure we need that? It's okay to change the color slightly if we have a similar in the codebase already.

File: packages/frontend/src/components/project/WarningBar.tsx [hunk 7]
```diff
@@ -9,6 +9,7 @@ export interface WarningBarProps {
   text: string
   isCritical: boolean
   className?: string
+  isArchived?: boolean
```
Comment (antooni) [inline packages/frontend/src/components/project/WarningBar.tsx:12]: This one is not needed any more

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+import { ArchivedIcon } from '../icons/symbols/ArchivedIcon'
+
+export interface ArchiveBarProps {
+  text: string
+}
+
+export function ArchivedBar({ text }: ArchiveBarProps) {
```
Comment (antooni) [inline packages/frontend/src/components/project/ArchivedBar.tsx:9]: Add storybook for this component

File: packages/frontend/src/components/header/DetailsHeader.tsx [hunk 9]
```diff
@@ -32,6 +34,9 @@ export function DetailsHeader(props: HeaderProps) {
           )}
           {props.title}
         </h1>
+        {props.isArchived && (
+          <ArchivedBar text="This project is archived and no longer maintained." />
```
Comment (antooni) [inline packages/frontend/src/components/header/DetailsHeader.tsx:38]: Add storybook for details with archive bar

**Review by antooni — CHANGES_REQUESTED** [review]
Add new UI elements to storybook

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+export interface ArchivedBarProps {
+  text: string
+}
+
+export function ArchivedBar({ text }: ArchivedBarProps) {
+  return (
+    <div
+      className={cx(
+        'mb-6 flex justify-center px-2.5 md:px-0',
+        'rounded-lg bg-gray-800 py-2.5',
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:15]:
```suggestion
        'mb-6 flex justify-center p-2.5 md:px-0',
        'rounded-lg bg-gray-800',
```

**Review by michalsidzej — APPROVED** [review]

**Review by michalsidzej — CHANGES_REQUESTED** [review]
The light mode implementation doesn't look quite right. 
Screenshot 2023-03-08 at 14 08 36

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+      className={cx(
+        'mb-6 flex justify-center py-2.5 md:px-0',
+        'rounded-lg bg-gray-200 dark:bg-gray-800',
+      )}
+    >
+      <span className="flex flex-row">
+        <span className="flex items-center">
+          <ArchivedIcon />
+        </span>
+        <span className="ml-2 text-base font-medium dark:text-white">
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:18]:
```suggestion
        <span className="ml-2 text-base font-medium">
```
Is it necessary? Text defaults to white on dark mode

File: packages/frontend/src/components/project/ArchivedBar.tsx [hunk 12]
```diff
@@ -3,23 +3,24 @@ import React from 'react'
...
-}
-
-export function ArchivedBar({ text }: ArchivedBarProps) {
+export function ArchivedBar() {
   return (
     <div
       className={cx(
-        'mb-6 flex justify-center p-2.5 md:px-0',
-        'rounded-lg bg-gray-800',
+        'mb-6 flex justify-center py-2.5 md:px-0',
```
Comment (michalsidzej) [inline packages/frontend/src/components/project/ArchivedBar.tsx:10]: I'm not sure what's your goal with `md:px-0` (px is 0 already, as it's never set to a value).

**Review by michalsidzej — APPROVED** [review]
LGTM

**Review by antooni — APPROVED** [review]

**Pull request merged**